app-modules/meeting-center/src/Managers/OutlookCalendarManager.php

```diff
@@ -31,65 +31,281 @@
 namespace Assist\MeetingCenter\Managers;
 
 use DateTime;
+use Carbon\Carbon;
+use DateTimeInterface;
+use Microsoft\Graph\Graph;
+use Microsoft\Graph\Model\Event;
+use Microsoft\Graph\Model\Attendee;
+use Microsoft\Graph\Model\BodyType;
+use Microsoft\Graph\Model\ItemBody;
+use Illuminate\Support\Facades\Http;
+use Microsoft\Graph\Model\EmailAddress;
+use Microsoft\Graph\Core\GraphConstants;
 use Assist\MeetingCenter\Models\Calendar;
+use GuzzleHttp\Exception\ClientException;
+use Microsoft\Graph\Model\DateTimeTimeZone;
 use Assist\MeetingCenter\Models\CalendarEvent;
 use Assist\MeetingCenter\Managers\Contracts\CalendarInterface;
 
 class OutlookCalendarManager implements CalendarInterface
 {
-    /**
-     * @return array<string, string>
-     */
     public function getCalendars(Calendar $calendar): array
     {
-        // TODO: Implement getCalendars() method.
+        $client = (new Graph())->setAccessToken($calendar->oauth_token);
 
-        return [];
-    }
+        $calendars = $client->createRequest('GET', '/me/calendars')
+            ->setReturnType(\Microsoft\Graph\Model\Calendar::class)
+            ->execute();
 
-    // https://github.com/microsoftgraph/msgraph-sample-phpapp/tree/main
-    // https://github.com/microsoftgraph/msgraph-sdk-php
-    // https://learn.microsoft.com/en-us/graph/api/resources/webhooks?view=graph-rest-1.0
+        return collect($calendars)->filter(fn (\Microsoft\Graph\Model\Calendar $item) => $item->getCanEdit())
+            ->mapWithKeys(fn (\Microsoft\Graph\Model\Calendar $item) => [$item->getId() => $item->getName()])
+            ->toArray();
+    }
 
     public function getEvents(Calendar $calendar, ?Datetime $start = null, ?Datetime $end = null, ?int $perPage = null): array
     {
-        // https://learn.microsoft.com/en-us/graph/api/user-list-calendars?view=graph-rest-1.0&tabs=http
-        // https://learn.microsoft.com/en-us/graph/api/user-list-events?view=graph-rest-1.0&tabs=http
+        $client = $this->makeClient($calendar);
+
+        $start = $start ?? now()->subYear()->startOfDay();
+
+        $end = $end ?? now()->addYear()->endOfDay();
```

Comment:
I did change my mind here while working on this one, I'll change Google to a year as well.

Since I was testing this with my real Canyon calendar, two years brings in a massive amount of events into the app and takes a little longer to sync.